regress/expected/agtype.out

```diff
@@ -2138,6 +2142,50 @@ SELECT agtype_to_int8(agtype_in('false'));
               0
 (1 row)
 
+--
+-- Test boolean to integer cast
+--
+SELECT agtype_to_int4(agtype_in('true'));
+ agtype_to_int4 
+----------------
+              1
+(1 row)
+
+SELECT agtype_to_int4(agtype_in('false'));
+ agtype_to_int4 
+----------------
+              0
+(1 row)
+
+SELECT agtype_to_int4(agtype_in('null'));
+ agtype_to_int4 
+----------------
+               
+(1 row)
+
+SELECT agtype_to_int4(agtype_in('1'));
+ agtype_to_int4 
+----------------
+              1
+(1 row)
+
+SELECT agtype_to_int4(agtype_in('1.45'));
+ agtype_to_int4 
+----------------
+              1
+(1 row)
+
+SELECT agtype_to_int4(agtype_in('1.444::numeric'));
+ agtype_to_int4 
+----------------
+              1
+(1 row)
+
+-- These should all fail
+SELECT agtype_to_int4(agtype_in('"string"'));
+ERROR:  invalid input syntax for integer: "string"
+SELECT agtype_to_int4(agtype_in('[1, 2, 3]'));
+ERROR:  cannot cast agtype array to type int
```

Comment:
You may also want to include objects in your tests, fx -

`SELECT agtype_to_int4(agtype_in('{name: "Jason", age: 34, gpa: 3.4}'));` 

or just numbers and characters -

`SELECT agtype_to_int4(1);`
`SELECT agtype_to_int4(3.1415);`
`SELECT agtype_to_int4(true);`
`SELECT agtype_to_int4('1');`

These are just examples.

Reply:
`SELECT agtype_to_int4(1);` and `SELECT agtype_to_int4(true);` termiantes the connection to server. The other two give error.

Reply:
If your changes cause the crash, or it is related to the area you are working in, I would say that you should address it with your PR. You can always state that you fixed issues with the code that caused crashes or whatever the case may be.

Reply:
@jrgemignani I have verified, the crash is not because of my changes but existed before. The server crashes for agtype_to_int2, agtype_to_int4, agtype_to_int8 for this input. 
I am looking for a solution to fix this, but I believe that dealing this bug here would make this PR very cluttered since there are many functions that need changes.
I can resolve this bug in another PR if that's fine to deal with these edge cases there.

Reply:
Okay. Create an issue for the crash and note that the regression tests that caused the failures will be addressed in the forthcoming patch.